src/common/DOMWorld.ts

```diff
@@ -457,70 +462,82 @@ export class DOMWorld {
     await handle.dispose();
   }
 
-  waitForSelector(
+  async waitForSelector(
     selector: string,
     options: WaitForSelectorOptions
   ): Promise<ElementHandle | null> {
-    return this._waitForSelectorOrXPath(selector, false, options);
-  }
-
-  waitForXPath(
-    xpath: string,
-    options: WaitForSelectorOptions
-  ): Promise<ElementHandle | null> {
-    return this._waitForSelectorOrXPath(xpath, true, options);
-  }
-
-  waitForFunction(
-    pageFunction: Function | string,
-    options: { polling?: string | number; timeout?: number } = {},
-    ...args: SerializableOrJSHandle[]
-  ): Promise<JSHandle> {
     const {
-      polling = 'raf',
+      visible: waitForVisible = false,
+      hidden: waitForHidden = false,
       timeout = this._timeoutSettings.timeout(),
     } = options;
-    return new WaitTask(
+    const polling = waitForVisible || waitForHidden ? 'raf' : 'mutation';
+    const title = `selector "${selector}"${
```

Comment:
Let's wrap the select in backticks instead of double quotes, since it's more likely to contain double quotes (e.g. attribute selectors) than backticks:

```suggestion
    const title = `selector \`${selector}\`${
```